rust/otel-arrow-rust/src/pdata/otlp/model/src/lib.rs

```diff
@@ -0,0 +1,297 @@
+		oneof("double", "f64", "any_value::Value::DoubleValue", None),
+		oneof("kvlist", "Vec<KeyValue>", "any_value::Value::KvlistValue", Some("KeyValueList::new")),
+		oneof("array", "Vec<AnyValue>", "any_value::Value::ArrayValue", Some("ArrayValue::new")),
+		oneof("bytes", "Vec<u8>", "any_value::Value::BytesValue", None),
+	    ],
+	},
+	OneofMapping{
+	    field: "opentelemetry.proto.metrics.v1.Metric.data",
+	    cases: vec![
+		oneof("sum", "Sum", "metric::Data::Sum", None),
+		oneof("gauge", "Gauge", "metric::Data::Gauge", None),
+		oneof("histogram", "Histogram", "metric::Data::Histogram", None),
+		oneof("exponential_histogram", "ExponentialHistogram", "metric::Data::ExponentialHistogram", None),
+		oneof("summary", "Summary", "metric::Data::Summary", None),
+	    ],
+	},
+	OneofMapping{
+	    field: "opentelemetry.proto.metrics.v1.NumberDataPoint.value",
+	    cases: vec![
+		oneof("int", "i64", "number_data_point::Value::AsInt", None),
+		oneof("double", "f64", "number_data_point::Value::AsDouble", None),
+	    ],
+	},
+	OneofMapping{
+	    field: "opentelemetry.proto.metrics.v1.Exemplar.value",
+	    cases: vec![
+		oneof("int", "i64", "exemplar::Value::AsInt", None),
+		oneof("double", "f64", "exemplar::Value::AsDouble", None),
+	    ],
+	},
+    ]
+});
+
+pub struct Override {
+    pub datatype: &'static str,
+    pub fieldtype: &'static str,
+}
+
+pub static FIELD_TYPE_OVERRIDES: LazyLock<HashMap<&'static str, Override>> = LazyLock::new(|| {
+    let mut m = HashMap::new();
+    m.insert(
+        "opentelemetry.proto.logs.v1.LogRecord.flags",
+        Override {
+            datatype: "LogRecordFlags",
+            fieldtype: "u32",
+        },
+    );
+    m.insert(
+        "opentelemetry.proto.trace.v1.Span.flags",
+        Override {
+            datatype: "SpanFlags",
+            fieldtype: "u32",
+        },
+    );
+    m.insert(
+        "opentelemetry.proto.trace.v1.Span.kind",
+        Override {
+            datatype: "span::SpanKind",
+            fieldtype: "i32",
+        },
+    );
+    m.insert(
+        "opentelemetry.proto.trace.v1.Status.code",
+        Override {
+            datatype: "status::StatusCode",
+            fieldtype: "i32",
+        },
+    );
+    m.insert(
+        "opentelemetry.proto.metrics.v1.Sum.aggregation_temporality",
+        Override {
+            datatype: "AggregationTemporality",
+            fieldtype: "i32",
+        },
+    );
+    m.insert(
+        "opentelemetry.proto.metrics.v1.Histogram.aggregation_temporality",
+        Override {
+            datatype: "AggregationTemporality",
+            fieldtype: "i32",
+        },
+    );
+    m.insert(
+        "opentelemetry.proto.metrics.v1.ExponentialHistogram.aggregation_temporality",
+        Override {
+            datatype: "AggregationTemporality",
+            fieldtype: "i32",
+        },
+    );
+    m
+});
+
+pub static DETAILS: LazyLock<Vec<Detail>> = LazyLock::new(|| {
+    vec![
```

Comment:
Any reason why we are using `vec` in some places instead of `Hashmap` like in `FIELD_TYPE_OVERRIDES`?

Reply:
I modified this file to be more consistent, using HashMap::from([...]) for all three model variables. Thanks!